hil/client/switch.py

```diff
@@ -23,7 +67,65 @@ def register(self, switch, subtype, *args):
 #       of the switches HIL will control and has read the
 #       HIL documentation to use appropriate flags to register
 #       it with HIL.
-        raise NotImplementedError
+        switch_api = "http://schema.massopencloud.org/haas/v0/switches/" + subtype
+        if switch_api in known_types:
+            try:
+                known_types[switch_api].validate(args)
+            except:
+                SchemaError("Bad Request")
```

Comment:
@zenhack @naved001 Another question starts after this line. So let'say we validate the `args` but we still need to construct the `switchinfo` so that we can pass all `args` to the REST API. Not sure whether the current validation mechanism would help us construct the `switchinfo`, any idea?

Reply:
the dictionary in Schema is the switchinfo we want. The `args` we are passing here `known_types[switch_api].validate(args)` isn't that `switchinfo` ? Though, you are right, we still need to build it before passing it.